app/views/auth_key_pair_cloud/_form.html.haml

```diff
@@ -1,44 +1,53 @@
+      %label.col-md-2.control-label{:for => 'name'}
         = _('Name')
       .col-md-8
-        = text_field_tag("name",
-                          @edit[:new][:name],
-                          :maxlength => 50,
-                          :class => "form-control",
-                          "data-miq_focus" => true,
-                          "data-miq_observe" => {:interval => '.5', :url => url}.to_json)
+        %input.form-control{:type            => 'text',
+                            :name            => 'name',
+                            :id              => 'name',
+                            'ng-model'       => 'keyPairModel.name',
+                            :maxlength       => MAX_NAME_LEN,
+                            :required        => true,
+                            :checkchange     => '',
+                            'auto-focus'     => true,
+                            'start-form-div' => 'start_form_div'}
+        %span.help-block{"ng-show" => "angularForm.name.$error.required"}
+          = _("Required")
     .form-group
-      %label.col-md-2.control-label
+      %label.col-md-2.control-label{"for" => 'public_key'}
         = _('Public Key (optional)')
       .col-md-8
-        = text_area_tag("public_key",
-                          @edit[:new][:public_key],
-                          :size => "50x10",
-                          :class => "form-control",
-                          "data-miq_observe" => {:interval => '.5', :url => url}.to_json)
-  .form-horizontal
-    .form-group
-      %label.col-md-2.control-label
-        = _('Provider')
-      .col-md-8
-        = select_tag("ems_id",
-                      options_for_select(@edit[:ems_choices].sort, @edit[:new][:ems_id]),
-                      :multiple => false,
-                      "data-miq_sparkle_on" => true,
-                      "data-miq_sparkle_off" => true,
-                      :class    => "selectpicker",
-                      "data-miq_observe" => {:interval => '.5', :url => url}.to_json)
-    :javascript
-       miqInitSelectPicker();
-       miqSelectPickerEvent("ems_id", "#{url}")
+        %textarea.form-control{:name           => 'public_key',
+                               :id             => 'public_key',
+                               'ng-model'      => 'keyPairModel.public_key',
+                               :rows           => 10}
+    .form-horizontal
+      .form-group
+        %label.col-md-2.control-label
+          = _('Provider')
+        .col-md-8
+          = select_tag('ems_id',
+                  '',
+                  'ng-model'                   => 'keyPairModel.ems',
+                  'ng-options'                 => 'ems.name for ems in ems_choices',
+                  :id                          => 'ems_id',
+                  :required                    => '',
+                  'pf-select'                  => true)
+          :javascript
+            miqInitSelectPicker();
```

Comment:
@dgutride Thought I made a comment here yesterday...but apparently not.
You can remove this `:javascript` block completely and instead use the directive `selectpicker-for-select-tag` for the `select_tag`. Also, you would not need the `pf-select` directive.